src/System.Numerics.Tensors/src/System/Numerics/Tensors/Experimental/IReadOnlySparseTensor.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Numerics.Tensors
+{
+    public interface IReadOnlySparseTensor<T> : IReadOnlyTensor<T>
+    {
+        long ValueCount { get; }
+
+        ReadOnlySpan<long> GetValueIndices(long valueIndex);
+
+        T GetValue(long valueIndex);
```

Comment:
I'm a bit confused at what `valueIndex` is in this API. Is it the linear index?  Or is the idea that I could go from `0` to `ValueCount` and call `GetValueIndices` and `GetValue` with that integer?

Reply:
To follow with that - having a dictionary backed sparse tensor, this may not be possible.  Which value in the dictionary is the first? the second?  etc.

---
In reply to: [265284272](https://github.com/dotnet/corefx/pull/36019#discussion_r265284272) [](ancestors = 265284272)

Reply:
The latter. You can either do `tensor.GetValue(0...ValueCount)` or `tensor[GetValueIndices(0...ValueCount)]`

Reply:
> To follow with that - having a dictionary backed sparse tensor, this may not be possible. 

What would be the key in this case? The linearized index?

Reply:
Yes, the linearized index, in the implementation @ericstj created:

https://github.com/dotnet/corefx/blob/fd4f870a9da036458309d0b833a2cd6ad5f354df/src/System.Numerics.Tensors/src/System/Numerics/Tensors/SparseTensor.cs#L9-L15

Reply:
I'm guessing that makes it more efficient for the `this[params long[] indices]` case, but slower for the case where you are performing operations on the tensors; but we should likely support that scenario in either case.

That would either involve dropping `GetValue(int valueIndex)` or requiring dictionary backed sparse tensors to potentially sort their keys in the `GetValue` implementation (the cost of which depends on whether or not `IDictionary<TKey, TValue>` naturally keeps their keys sorted).